__tests__/cmds/openapi/reduce.test.ts

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable no-console */
+import fs from 'fs';
+
+import chalk from 'chalk';
+import prompts from 'prompts';
+
+import OpenAPIReduceCommand from '../../../src/cmds/openapi/reduce';
+
+const reducer = new OpenAPIReduceCommand();
+
+const successfulReduction = specPath => `${specPath} has been reduced to output.json! 🤏`;
+
+const testWorkingDir = process.cwd();
+
+let consoleInfoSpy;
+const getCommandOutput = () => consoleInfoSpy.mock.calls.join('\n\n');
+
+describe('rdme openapi:reduce', () => {
+  beforeEach(() => {
+    jest.mock('fs');
+
+    consoleInfoSpy = jest.spyOn(console, 'info').mockImplementation();
+  });
+
+  afterEach(() => {
+    consoleInfoSpy.mockRestore();
+
+    process.chdir(testWorkingDir);
+
+    jest.clearAllMocks();
+  });
+
+  describe('reducing', () => {
+    describe('by tag', () => {
+      it.each([
+        ['OpenAPI 3.0', 'json', '3.0'],
+        ['OpenAPI 3.0', 'yaml', '3.0'],
+        ['OpenAPI 3.1', 'json', '3.1'],
+        ['OpenAPI 3.1', 'yaml', '3.1'],
+      ])('should support reducing a %s definition (format: %s)', async (_, format, specVersion) => {
+        const spec = require.resolve(`@readme/oas-examples/${specVersion}/${format}/petstore.${format}`);
+
+        let reducedSpec;
+        fs.writeFileSync = jest.fn((f, d) => {
```

Comment:
Cleaner to mock this out than writing, reading, and deleting the test file on the filesystem.

Reply:
writing out jest mocks

Reply:
jest mocks freak me out after #570 but this approach makes sense lol